### ISSUE TITLE
H5Dataset object becomes None

### ISSUE THREAD

**Issue by rwegener2** [ICESat2-SlideRule/h5coro#9]
### Goal
I am working on extracting the variable data and metadata from a group of a dataset, but every data variable I access returns None for the data. 

### Code
The code I use to access the dataset (no problems in this portion):
```python
import earthaccess
from h5coro import h5coro, s3driver

s3url_atl03 = 'nsidc-cumulus-prod-protected/ATLAS/ATL03/006/2019/11/30/ATL03_20191130112041_09860505_006_01.h5'

# authenticate
auth = earthaccess.login()
s3_creds = auth.get_s3_credentials(daac='NSIDC')

h5obj = h5coro.H5Coro(s3url_atl03, 
                      s3driver.S3Driver,
                      credentials={"aws_access_key_id": s3_creds["accessKeyId"],
                                 "aws_secret_access_key": s3_creds["secretAccessKey"],
                                 "aws_session_token": s3_creds["sessionToken"], })
```
I then access the variables using:
```python
group='/gt1l/heights'
# loop through that dictionary of datasets to generate a list of groups/variables and attrs
variables, attributes = h5obj.listGroup(group, w_attr=True, w_inspect=True)
dataarray_dicts = {}
coords = {}
for variable, results in variables.items():
    # pull out data
    print('------ PROCESSING VARIABLE -----', variable)
    var_data = h5obj.readDatasets(datasets=[variable[1:]], block=True)
    print('looking at readDatasets output #1', var_data)

    # format data into xarray-compatible dictionaries
    coordinate_names = [os.path.join(group, var) for var in ['lon_ph', 'lat_ph', 'delta_time',]]
    if variable in coordinate_names:  # put the coordinates in their own array
        coords[variable.split('/')[-1]] = var_data[variable[1:]]
    elif variable in os.path.join(group, 'signal_conf_ph'):
        # ignore the 2d variable manually for now
        pass
    else:  # everything else is 1d variable data
        print('using key ', variable[1:], 'on keys list', var_data.keys())
        print('looking at readDatasets output #2', var_data[variable[1:]])
        dataarray_dicts[variable.split('/')[-1]] = ("delta_time", var_data[variable[1:]])
```
There are multiple print statements in there, as I try to isolate where the data is disappearing. You'll notice in the output (below) that the first print statement is totally normal (returns an H5Metadata object), while the second one returns None. This happens for all the variables in the loop, but I just put one variable here.
```python
------ PROCESSING VARIABLE ----- /gt1l/heights/dist_ph_across
looking at readDatasets output #1 { "gt1l/heights/dist_ph_across": <h5coro.h5dataset.H5Dataset object at 0x7f85a4ae73d0> }
using key  gt1l/heights/dist_ph_across on keys list dict_keys(['gt1l/heights/dist_ph_across'])
looking at readDatasets output #2 None
```
I know there isn't a real question in here; for now I'm just looking for a sanity check - Is this confusing to anyone else? I feel like data I had is totally disappearing. Am I missing something? Thanks for any perspectives 🙏🏻

**Comment by jpswinski** [comment]
@rwegener2 you uncovered a bug in the caching logic in the H5Datset class.  I've pushed a fix to`main` [af4bfcf](https://github.com/ICESat2-SlideRule/h5coro/commit/af4bfcf5fce69f3f50060489bd070b7b3b9facd2) and I reran your script above and I think it fixes it.  Can you confirm?

**Comment by rwegener2** [comment]
Runs smoothly now. Thanks @jpswinski! 🙌🏻
